machine/machine_core/machine_virtual.py

```diff
@@ -320,12 +321,14 @@ def execute(*args):
             self.message(*args)
             return subprocess.check_call(args)
 
-        image_to_use = self.image_file
         if not self.maintain:
-            (unused, self._transient_image) = tempfile.mkstemp(suffix='.qcow2', prefix='cockpit-', dir=self.run_dir)
+            self._transient_image = tempfile.NamedTemporaryFile(suffix='.qcow2', prefix='cockpit-', dir=self.run_dir)
```

Comment:
:+1: I remember when we wrote that, NamedTemporaryFile wasn't a thing yet.

Reply:
:)